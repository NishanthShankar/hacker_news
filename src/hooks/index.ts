import { useQuery } from 'react-query'

// Mock api
const APIS: { [key: string]: string } = {
  topstories:
    'https://hacker-news.firebaseio.com/v0/topstories.json?print=pretty',
  ask: 'https://hacker-news.firebaseio.com/v0/askstories.json?print=pretty',
  jobs: 'https://hacker-news.firebaseio.com/v0/jobstories.json?print=pretty',
  show: 'https://hacker-news.firebaseio.com/v0/showstories.json?print=pretty'
}

// const G_KEY = 'AIzaSyALeDOcqLyxVSRQxMRMhZMRc7WDv4cgDfk'
// const api = `https://www.googleapis.com/youtube/v3/search?part=snippet&type=video&maxResults=50&key=${G_KEY}`

export const fetchTopSearch = (id: string) => async (): Promise<any> => {
  const URL = APIS[id] || APIS.topstories
  const res = await fetch(URL)
  return await res.json()
}

export const useSearch = (id: string): any => {
  return useQuery<any, Error>(id, fetchTopSearch(id))
}
