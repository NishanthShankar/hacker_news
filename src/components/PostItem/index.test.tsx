import Enzyme, { mount } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import Component from './index'
import { useSearch } from 'hooks/'
import toJson from 'enzyme-to-json'
import { QueryClient, QueryClientProvider } from 'react-query'
Enzyme.configure({ adapter: new Adapter() })

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      refetchOnWindowFocus: false
    }
  }
})
// Solves TypeScript Errors
const mockedUseSearch = useSearch as jest.Mock<any>

// Mock the module
jest.mock('../../hooks/')

const Element = (id: string): JSX.Element => (
  <QueryClientProvider client={queryClient}>
    <Component id={id} />
  </QueryClientProvider>
)

describe('<PostItem />', () => {
  beforeEach(() => {
    mockedUseSearch.mockImplementation(() => ({ isLoading: true }))
  })
  afterEach(() => {
    jest.clearAllMocks()
  })
  it('renders', () => {
    mockedUseSearch.mockImplementation(() => ({
      isLoading: false,
      data: {
        by: 'avestura',
        descendants: 81,
        id: 30965352,
        kids: [30966907, 30966008, 30966814],
        score: 325,
        time: 1649479929,
        title: 'Persepolis of ancient Persia rendered in WebGL',
        type: 'story',
        url: 'https://persepolis.getty.edu'
      }
    }))
    const tree = mount(Element('30965352'))
    expect(toJson(tree)).toMatchSnapshot()
  })
})
