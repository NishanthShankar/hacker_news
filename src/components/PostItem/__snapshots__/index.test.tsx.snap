// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostItem /> renders 1`] = `
<QueryClientProvider
  client={
    QueryClient {
      "defaultOptions": Object {
        "queries": Object {
          "refetchOnWindowFocus": false,
        },
      },
      "mutationCache": MutationCache {
        "config": Object {},
        "listeners": Array [],
        "mutationId": 0,
        "mutations": Array [],
      },
      "mutationDefaults": Array [],
      "queryCache": QueryCache {
        "config": Object {},
        "listeners": Array [],
        "queries": Array [
          Query {
            "abortSignalConsumed": false,
            "cache": [Circular],
            "cacheTime": 300000,
            "defaultOptions": undefined,
            "gcTimeout": undefined,
            "hadObservers": true,
            "initialState": Object {
              "data": undefined,
              "dataUpdateCount": 0,
              "dataUpdatedAt": 0,
              "error": null,
              "errorUpdateCount": 0,
              "errorUpdatedAt": 0,
              "fetchFailureCount": 0,
              "fetchMeta": null,
              "isFetching": false,
              "isInvalidated": false,
              "isPaused": false,
              "status": "idle",
            },
            "meta": undefined,
            "observers": Array [
              QueryObserver {
                "client": [Circular],
                "currentQuery": [Circular],
                "currentQueryInitialState": Object {
                  "data": undefined,
                  "dataUpdateCount": 0,
                  "dataUpdatedAt": 0,
                  "error": null,
                  "errorUpdateCount": 0,
                  "errorUpdatedAt": 0,
                  "fetchFailureCount": 0,
                  "fetchMeta": null,
                  "isFetching": false,
                  "isInvalidated": false,
                  "isPaused": false,
                  "status": "idle",
                },
                "currentRefetchInterval": false,
                "currentResult": Object {
                  "data": undefined,
                  "dataUpdatedAt": 0,
                  "error": null,
                  "errorUpdatedAt": 0,
                  "failureCount": 0,
                  "isError": false,
                  "isFetched": false,
                  "isFetchedAfterMount": false,
                  "isFetching": true,
                  "isIdle": false,
                  "isLoading": true,
                  "isLoadingError": false,
                  "isPlaceholderData": false,
                  "isPreviousData": false,
                  "isRefetchError": false,
                  "isRefetching": false,
                  "isStale": true,
                  "isSuccess": false,
                  "refetch": [Function],
                  "remove": [Function],
                  "status": "loading",
                },
                "currentResultOptions": Object {
                  "_defaulted": true,
                  "optimisticResults": true,
                  "queryFn": [Function],
                  "queryHash": "[\\"item\\",\\"30965352\\"]",
                  "queryKey": Array [
                    "item",
                    "30965352",
                  ],
                  "refetchOnWindowFocus": false,
                },
                "currentResultState": Object {
                  "data": undefined,
                  "dataUpdateCount": 0,
                  "dataUpdatedAt": 0,
                  "error": null,
                  "errorUpdateCount": 0,
                  "errorUpdatedAt": 0,
                  "fetchFailureCount": 0,
                  "fetchMeta": null,
                  "isFetching": true,
                  "isInvalidated": false,
                  "isPaused": false,
                  "status": "loading",
                },
                "listeners": Array [
                  [Function],
                ],
                "options": Object {
                  "_defaulted": true,
                  "optimisticResults": true,
                  "queryFn": [Function],
                  "queryHash": "[\\"item\\",\\"30965352\\"]",
                  "queryKey": Array [
                    "item",
                    "30965352",
                  ],
                  "refetchOnWindowFocus": false,
                },
                "previousQueryResult": undefined,
                "previousSelectError": null,
                "refetch": [Function],
                "refetchIntervalId": undefined,
                "remove": [Function],
                "staleTimeoutId": undefined,
                "trackedProps": Array [],
              },
            ],
            "options": Object {
              "_defaulted": true,
              "optimisticResults": true,
              "queryFn": [Function],
              "queryHash": "[\\"item\\",\\"30965352\\"]",
              "queryKey": Array [
                "item",
                "30965352",
              ],
              "refetchOnWindowFocus": false,
            },
            "promise": Promise {},
            "queryHash": "[\\"item\\",\\"30965352\\"]",
            "queryKey": Array [
              "item",
              "30965352",
            ],
            "retryer": Retryer {
              "abort": [Function],
              "cancel": [Function],
              "cancelRetry": [Function],
              "continue": [Function],
              "continueRetry": [Function],
              "failureCount": 0,
              "isPaused": false,
              "isResolved": false,
              "isTransportCancelable": false,
              "promise": Promise {},
            },
            "revertState": Object {
              "data": undefined,
              "dataUpdateCount": 0,
              "dataUpdatedAt": 0,
              "error": null,
              "errorUpdateCount": 0,
              "errorUpdatedAt": 0,
              "fetchFailureCount": 0,
              "fetchMeta": null,
              "isFetching": false,
              "isInvalidated": false,
              "isPaused": false,
              "status": "idle",
            },
            "state": Object {
              "data": undefined,
              "dataUpdateCount": 0,
              "dataUpdatedAt": 0,
              "error": null,
              "errorUpdateCount": 0,
              "errorUpdatedAt": 0,
              "fetchFailureCount": 0,
              "fetchMeta": null,
              "isFetching": true,
              "isInvalidated": false,
              "isPaused": false,
              "status": "loading",
            },
          },
        ],
        "queriesMap": Object {
          "[\\"item\\",\\"30965352\\"]": Query {
            "abortSignalConsumed": false,
            "cache": [Circular],
            "cacheTime": 300000,
            "defaultOptions": undefined,
            "gcTimeout": undefined,
            "hadObservers": true,
            "initialState": Object {
              "data": undefined,
              "dataUpdateCount": 0,
              "dataUpdatedAt": 0,
              "error": null,
              "errorUpdateCount": 0,
              "errorUpdatedAt": 0,
              "fetchFailureCount": 0,
              "fetchMeta": null,
              "isFetching": false,
              "isInvalidated": false,
              "isPaused": false,
              "status": "idle",
            },
            "meta": undefined,
            "observers": Array [
              QueryObserver {
                "client": [Circular],
                "currentQuery": [Circular],
                "currentQueryInitialState": Object {
                  "data": undefined,
                  "dataUpdateCount": 0,
                  "dataUpdatedAt": 0,
                  "error": null,
                  "errorUpdateCount": 0,
                  "errorUpdatedAt": 0,
                  "fetchFailureCount": 0,
                  "fetchMeta": null,
                  "isFetching": false,
                  "isInvalidated": false,
                  "isPaused": false,
                  "status": "idle",
                },
                "currentRefetchInterval": false,
                "currentResult": Object {
                  "data": undefined,
                  "dataUpdatedAt": 0,
                  "error": null,
                  "errorUpdatedAt": 0,
                  "failureCount": 0,
                  "isError": false,
                  "isFetched": false,
                  "isFetchedAfterMount": false,
                  "isFetching": true,
                  "isIdle": false,
                  "isLoading": true,
                  "isLoadingError": false,
                  "isPlaceholderData": false,
                  "isPreviousData": false,
                  "isRefetchError": false,
                  "isRefetching": false,
                  "isStale": true,
                  "isSuccess": false,
                  "refetch": [Function],
                  "remove": [Function],
                  "status": "loading",
                },
                "currentResultOptions": Object {
                  "_defaulted": true,
                  "optimisticResults": true,
                  "queryFn": [Function],
                  "queryHash": "[\\"item\\",\\"30965352\\"]",
                  "queryKey": Array [
                    "item",
                    "30965352",
                  ],
                  "refetchOnWindowFocus": false,
                },
                "currentResultState": Object {
                  "data": undefined,
                  "dataUpdateCount": 0,
                  "dataUpdatedAt": 0,
                  "error": null,
                  "errorUpdateCount": 0,
                  "errorUpdatedAt": 0,
                  "fetchFailureCount": 0,
                  "fetchMeta": null,
                  "isFetching": true,
                  "isInvalidated": false,
                  "isPaused": false,
                  "status": "loading",
                },
                "listeners": Array [
                  [Function],
                ],
                "options": Object {
                  "_defaulted": true,
                  "optimisticResults": true,
                  "queryFn": [Function],
                  "queryHash": "[\\"item\\",\\"30965352\\"]",
                  "queryKey": Array [
                    "item",
                    "30965352",
                  ],
                  "refetchOnWindowFocus": false,
                },
                "previousQueryResult": undefined,
                "previousSelectError": null,
                "refetch": [Function],
                "refetchIntervalId": undefined,
                "remove": [Function],
                "staleTimeoutId": undefined,
                "trackedProps": Array [],
              },
            ],
            "options": Object {
              "_defaulted": true,
              "optimisticResults": true,
              "queryFn": [Function],
              "queryHash": "[\\"item\\",\\"30965352\\"]",
              "queryKey": Array [
                "item",
                "30965352",
              ],
              "refetchOnWindowFocus": false,
            },
            "promise": Promise {},
            "queryHash": "[\\"item\\",\\"30965352\\"]",
            "queryKey": Array [
              "item",
              "30965352",
            ],
            "retryer": Retryer {
              "abort": [Function],
              "cancel": [Function],
              "cancelRetry": [Function],
              "continue": [Function],
              "continueRetry": [Function],
              "failureCount": 0,
              "isPaused": false,
              "isResolved": false,
              "isTransportCancelable": false,
              "promise": Promise {},
            },
            "revertState": Object {
              "data": undefined,
              "dataUpdateCount": 0,
              "dataUpdatedAt": 0,
              "error": null,
              "errorUpdateCount": 0,
              "errorUpdatedAt": 0,
              "fetchFailureCount": 0,
              "fetchMeta": null,
              "isFetching": false,
              "isInvalidated": false,
              "isPaused": false,
              "status": "idle",
            },
            "state": Object {
              "data": undefined,
              "dataUpdateCount": 0,
              "dataUpdatedAt": 0,
              "error": null,
              "errorUpdateCount": 0,
              "errorUpdatedAt": 0,
              "fetchFailureCount": 0,
              "fetchMeta": null,
              "isFetching": true,
              "isInvalidated": false,
              "isPaused": false,
              "status": "loading",
            },
          },
        },
      },
      "queryDefaults": Array [],
      "unsubscribeFocus": [Function],
      "unsubscribeOnline": [Function],
    }
  }
>
  <PostItem
    id="30965352"
  >
    <View>
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <View>
          <div
            style={
              Object {
                "display": "flex",
                "flexDirection": "column",
              }
            }
          />
        </View>
        <View>
          <div
            style={
              Object {
                "display": "flex",
                "flexDirection": "column",
              }
            }
          />
        </View>
      </div>
    </View>
  </PostItem>
</QueryClientProvider>
`;
